pyro_risks/evaluate.py

```diff
@@ -0,0 +1,203 @@
+
+
+def evaluate_pipeline(
+    X: pd.DataFrame,
+    y: pd.Series,
+    pipeline: Optional[Union[pp.Pipeline, str]],
+    threshold: np.float64,
+    prefix: Optional[str] = None,
+    destination: Optional[str] = None,
+):
+    """
+    Build and save binary classification evaluation reports.
+
+    Args:
+        X: Training dataset features pd.DataFrame.
+        y: Training dataset target pd.Series.
+        pipeline: imbalanced-learn preprocessing pipeline or path to pipeline. Defaults to None.
+        threshold: Classification pipeline optimal threshold.
+        prefix: Classification reports prefix i.e. pipeline name. Defaults to None.
+        destination: Folder where the report should be saved. Defaults to ``METADATA_REGISTRY``.
+    """
+    # setup
+    _, X_test, _, y_test = train_test_split(
+        X, y, test_size=cfg.TEST_SIZE, random_state=cfg.RANDOM_STATE
+    )
+
+    if not isinstance(pipeline, pp.Pipeline):
+        pipeline = joblib.load(pipeline)
+
+    y_proba = pipeline.predict_proba(X_test)
+
+    def predict(x):
+        return 1 if x > threshold else 0
+
+    vpredict = np.vectorize(predict)
+    vdiscretizer = np.vectorize(discretizer)
+
+    y_pred = vpredict(y_proba[:, 1])
+    y_test = vdiscretizer(y_test)
+
+    save_classification_reports(
+        y_true=y_test, y_pred=y_pred, prefix=prefix, destination=destination
+    )
+
+    save_classification_plots(
+        y_true=y_test,
+        y_proba=y_proba[:, 1],
+        threshold=threshold,
+        prefix=prefix,
+        destination=destination,
+    )
+
+
+def main(args):
+    usecols = [cfg.DATE_VAR, cfg.ZONE_VAR, cfg.TARGET] + cfg.PIPELINE_ERA5T_VARS
+    pipeline_vars = [cfg.DATE_VAR, cfg.ZONE_VAR] + cfg.PIPELINE_ERA5T_VARS
+    df = pd.read_csv(cfg.ERA5T_VIIRS_PIPELINE, usecols=usecols)
+    df["day"] = df["day"].apply(
+        lambda x: datetime.strptime(str(x), "%Y-%m-%d") if not pd.isnull(x) else x
+    )
+    X = df[pipeline_vars]
+    y = df[cfg.TARGET]
+    evaluate_pipeline(
+        X=X,
+        y=y,
+        pipeline=args.pipeline,
+        threshold=args.threshold,
+        prefix=args.prefix,
+        destination=args.destination,
+    )
+
+
+def parse_args(args):
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description="Pyrorisks Classification Pipeline Evaluation",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("--pipeline", help="Pipeline location path.")
+    parser.add_argument(
+        "--threshold", type=float, help="Classification pipeline optimal threshold."
+    )
+    parser.add_argument(
+        "--prefix",
+        default=None,
+        help="Classification reports prefix i.e. pipeline name.",
+    )
+    parser.add_argument(
+        "--destination", default=None, help="Folder where the report should be saved."
+    )
+    return parser.parse_args(args)
+
+
+if __name__ == "__main__":
+
+    args = parse_args(sys.argv[1:])
+    main(args)
```

Comment:
Just my point of view but: if this is a library, scripts should be clearly separated from modules. We would have to move train.py as well, but I suggest to keep everything in the `pyro_risks` folder as a module and not a script

Reply:
(You could add the script part in the `scripts` folder)

Reply:
Thanks for the review! I understand what you are proposing but I also think that having entry points directly in the package is valuable. 

A middle ground could be to refactor all the scripts into a single entry and to built a CLI with nested command. We could then install the package from GitHub or PyPi and directly run: 
 
```bash
pyrorisks evaluate --destination ./metadata-registry
pyrorisks train --destination ./model-registry
```

The package tree could look like that which is better than what I am doing now for sure 

```bash
├── config
│   ├── __init__.py
│   ├── datasets.py
│   ├── logger.py
│   └── models.py
├── models
│   ├── __init__.py
│   ├── pipelines.py
│   ├── transformers.py
│   └── utils.py
├── pipelines
│   ├── load.py
│   ├── evaluate.py
│   ├── predict.py
│   ├── train.py
├──  main.py
├──  version.py
```


I will open an issue for that ! 

In the meantime these entry points at the end of the files are really helpful and helps with trying out machine learning orchestration frameworks as mentioned in the issue #37 and I have proposals that I want to discuss soon with the team in another issue 😄 

Let me know if It sounds right to you !